### PR TITLE
Open drawer for programs

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [ ] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [ ] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Partially addresses #2152

#### What's this PR do?
Opens the `LearningResourceDrawer` when a program search result/card is clicked and displays the relevant info.

#### How should this be manually tested?
- import some programs if you don't have any yet:
  ```
   from course_catalog.tasks import *
   get_xpro_data()
   sync_and_upload_edx_data()
   get_micromasters_data()
  ```
  you'll need to set some .env variables for the above including:
  ```
  MICROMASTERS_CATALOG_API_URL=http://docker.for.mac.host.internal:8079/api/v0/catalog/
  XPRO_CATALOG_API_URL=http://docker.for.mac.host.internal:8053/api/programs/
  ```
- Search for programs at `/courses/search`.  Click one of the results.  The drawer should open with the relevant info and it should look roughly like the [design](https://projects.invisionapp.com/d/main#/console/17477251/376322443/preview), but many fields will be missing due to the info not existing for programs (length, effort, video transcript, topics, instructors, etc).
- ~The image will be broken for xpro courses because the xpro API lists them as relative URL's (will be fixed soon)~.
- If there is a link to a Micromasters program, it won't work either because it is imported as a relative URL (https://github.com/mitodl/micromasters/issues/4410)

#### Screenshots (if appropriate)
Screen Shot 2019-09-27 at 10 35 37 AM

Screen Shot 2019-09-27 at 10 33 52 AM

### CODE REVIEW

**Comment by mbertrand** [comment]
@ferdi @pdpinch @abdulkdawson screenshots are in the description

**Comment by Ferdi** [comment]
Do we have start_date for xpro/MM ?
Do we have instructors for xpro / MM ?
Why there is not "Take course on" button in the first screenshot
Can you separate "take course on" from the provider like in the design?
replace "Number of courses : 5 courses in program" to "Number of courses : 5" or "Number of courses in program: 5"

**Comment by mbertrand** [comment]
> Do we have start_date for xpro/MM ?
Not yet, it's being worked on for xpro programs, not sure about MM
> Do we have instructors for xpro / MM ?
No, instructors are populated for course runs only.
> Why there is not "Take course on" button in the first screenshot
There is no URL being imported yet for xPro programs.
> Can you separate "take course on" from the provider like in the design?
Sure, but my assumption is that styling changes for the drawer would be in a different PR.  
> replace "Number of courses : 5 courses in program" to "Number of courses : 5" or "Number of courses in program: 5"
OK

**Comment by mbertrand** [comment]
@ferdi I changed the code to display all unique instructors from all course runs included in a program.

**Comment by mbertrand** [comment]
@ferdi regarding "Can you separate "take course on" from the provider like in the design?" - presumably this would apply to courses, bootcamps too?  I would need logos for all the "offered by" options (Micromasters, xPro, etc) - seems like this should be handled in a separate PR.

**Comment by mbertrand** [comment]
@ferdi @pdpinch @abdulkdawson I just updated the screenshots

**Comment by Ferdi** [comment]
@mbertrand 

I suggest we have the providers APIs expose the list of instructors and other metadata. Each of those (MM, xpro etc.) can use their own logic to construct the data, and Open just consumes what comes from these api-s.

**Comment by mbertrand** [comment]
@ferdi ok I will create an issue for that in xpro and micromasters.  Should I remove the new code here that extracts the instructor list from all the course runs, or leave as is until the above is completed?

**Comment by Ferdi** [comment]
@mbertrand up to you.

File: course_catalog/serializers.py [hunk 1]
```diff
@@ -399,6 +399,7 @@ class ProgramSerializer(serializers.ModelSerializer, FavoriteSerializerMixin):
 
     items = ProgramItemSerializer(many=True, allow_null=True)
     topics = CourseTopicSerializer(read_only=True, many=True, allow_null=True)
+    prices = CoursePriceSerializer(read_only=True, many=True, allow_null=True)
```
Comment (noisecapella) [inline course_catalog/serializers.py]: Do we need to expose all of the prices via REST API? If we only need the most recent one we should expose that specifically.

File: static/js/lib/learning_resources.js [hunk 2]
```diff
@@ -143,16 +147,20 @@ export const resourceLabel = (resource: string) => {
...
 
-export const maxPrice = (courseRun: ?CourseRun) => {
-  const prices = courseRun && courseRun.prices ? courseRun.prices : []
+export const maxPrice = (prices: Array<CoursePrice>) => {
+  if (emptyOrNil(prices)) {
+    return null
+  }
   const price = Math.max(...prices.map(price => price.price))
-  return price > 0 ? `$${price}` : "Free"
+  return price > 0 ? `$${price.toFixed(2)}` : "Free"
```
Comment (noisecapella) [inline static/js/lib/learning_resources.js]: Can we make a separate function for this? Maybe similar to https://github.com/mitodl/mitxpro/blob/master/static/js/lib/ecommerce.js#L39

File: course_catalog/serializers.py (hunk 1 above)
Comment (mbertrand) [reply course_catalog/serializers.py]: Program prices don't have any dates (besides created/modified), and for xpro at least there should be only one anyway.  I don't think the micromasters catalog API returns any prices at all yet.

File: static/js/lib/learning_resources.js (hunk 2 above)
Comment (mbertrand) [reply static/js/lib/learning_resources.js]: Done

File: static/js/components/LearningResourceDrawer.js [hunk 3]
```diff
@@ -105,6 +108,10 @@ const getObject = createSelector(
       return querySelectors.isFinished(queries, bootcampRequest(objectId))
         ? bootcamps[objectId]
         : null
+    case LR_TYPE_PROGRAM:
```
Comment (noisecapella) [inline static/js/components/LearningResourceDrawer.js]: It looks like this case isn't covered by any tests. Can you add a test for it?

**Comment by mbertrand** [comment]
@pdpinch do you want to review the screenshots before I merge?

**Comment by mbertrand** [comment]
Oops sorry @pdpinch just noticed you're on vacation, so never mind.  @ferdi do the screenshots look okay to you?

**Pull request merged**